rules/Php70/Rector/StaticCall/StaticCallOnNonStaticToInstanceCallRector.php

```diff
@@ -173,6 +178,15 @@ private function shouldSkip(string $methodName, string $className, StaticCall $s
             return true;
         }
 
+        //Don't process static called methods from parent classes
+        $thisType = $scope->getVariableType('this');
+        if ($thisType instanceof ThisType) {
+            $thisClassReflection = $thisType->getClassReflection();
+            if ($thisClassReflection->isSubclassOf($className)) {
+                return true;
+            }
+        }
```

Comment:
this should only verify that className is abstract, so should be something like:

```suggestion
       // cannot instantiate abstract class
       if ($classReflection->isAbstract()) {
           return true;
       }
```

Reply:
Hi @samsonasik, you're right, there is not a real problem by the refactoring of
https://getrector.org/demo/24102d31-d518-4e51-8470-e3f93073470b

I have additionally looked what happened if class `A` requires constructor arguments, but that handled correctly by skipping the transformation: https://getrector.org/demo/2079937b-bb5d-48b3-96ed-c91a6d4d06ef

But I have a legacy code where the constructor has optionally parameter and the parent constructor must skip. In this case, Rector will break the code
https://getrector.org/demo/d8d9037d-1af5-4477-aeaa-7947d0089fdc

What do you think, should we skip magic methods or is this a kind of special use case and I should skip the rule completely for this project?